framework/platform/bpl/uci/db/bpl_db.cpp

```diff
@@ -17,45 +19,100 @@
 namespace beerocks {
 namespace bpl {
 
+constexpr char DB_FILE[] = "prplmesh_db";
+
 bool db_has_entry(const std::string &entry_type, const std::string &entry_name)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return false;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+    return db::uci_section_exists(DB_FILE, entry_type, entry_name);
 }
 
 bool db_add_entry(const std::string &entry_type, const std::string &entry_name,
                   const std::unordered_map<std::string, std::string> &params)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return true;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+
+    // Check if entry of the same name and type exist.
+    if (!db::uci_section_exists(DB_FILE, entry_type, entry_name)) {
+        LOG(DEBUG) << "Entry " << entry_name << " does not exist! Create new one.";
+        if (!db::uci_add_section(DB_FILE, entry_type, entry_name)) {
+            LOG(ERROR) << "Failed to create entry " << entry_name << "!";
+            return false;
+        }
+    }
+    // Update the existing entry
+    LOG(DEBUG) << "Update entry " << entry_name << " with new values.";
+    return db::uci_set_section(DB_FILE, entry_type, entry_name, params);
 }
 
 bool db_set_entry(const std::string &entry_type, const std::string &entry_name,
                   const std::unordered_map<std::string, std::string> &params)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return true;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+
+    if (!db::uci_section_exists(DB_FILE, entry_type, entry_name)) {
+        LOG(ERROR) << "Entry " << entry_name << " not found, or exists under a diffrent type!";
+        return false;
+    }
+    // Update the new/existing entry
+    LOG(DEBUG) << "Update entry " << entry_name << " with new values.";
+    return db::uci_set_section(DB_FILE, entry_type, entry_name, params);
 }
 
 bool db_get_entry(const std::string &entry_type, const std::string &entry_name,
                   std::unordered_map<std::string, std::string> &params)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
+    LOG(TRACE) << entry_type << ":" << entry_name;
+
+    if (params.size() == 0) {
+        // params map is empty, getting all parameters
+        return db::uci_get_section(DB_FILE, entry_type, entry_name, params);
+    }
+    for (auto &param : params) {
+        // params map is not empty, getting selected parameters
+        if (!db::uci_get_option(DB_FILE, entry_type, entry_name, param.first, param.second)) {
+            LOG(ERROR) << "Failed to get " << param.first;
+            // TODO:
+            return false;
+        }
+    }
```

Comment:
see my comment above regarding empty-name

Reply:
Added check for `!entry_name.empty()` in all functions and in `db_add_entry` for checked for `!entry_type.empty()` since the add function must have a type.